DrinkingPub/Order.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vsite.Oom.DrinkingPub
+{
+    public class Order
+    {
+        private readonly Dictionary<string, int> items;
+
+        public Order()
+        {
+            items = new Dictionary<string, int>();
+        }
+
+        public void AddItem(string itemName, int quantity)
+        {
+            if (items.ContainsKey(itemName))
+            {
+                items[itemName] += quantity;
+            }
+            else
+            {
+                items[itemName] = quantity;
+            }
+        }
+
+        public Dictionary<string, int> Items => items;
```

Comment:
Budući da ovo svojstvo vraća referencu na član, preko tog svojstva može bilo tko mijenjati sadržaj rječnika `items`. To možete jednostavno provjeriti tako da u metodu `Table.TotalToPay` ispred petlje foreach dodate naredbu:
`currentOrder.AddItem("Vinjak", 5);`
Da onemogućite promjene izvan klase `Order`, svojstvo definirajte kao:
`public IEnumerable<KeyValuePair<string, int>> Items => items.AsEnumerable();`